src/clj/puppetlabs/trapperkeeper/services/watcher/filesystem_watch_core.clj

```diff
@@ -125,12 +150,12 @@
    shutdown-fn :- IFn]
   (future
     (let [stopped? (atom false)]
-      (while (not @stopped?)
-        (try
-          (let [[watch-key events] (retrieve-events watcher)]
-            (when-not (empty? events)
-              (process-events! watcher watch-key events shutdown-fn)))
-         (catch ClosedWatchServiceException e
-           (reset! stopped? true)
-           (log/info (trs "Closing watcher {0}" watcher))))))))
+      (shutdown-fn #(while (not @stopped?)
+                     (try
+                       (let [events (retrieve-events watcher)]
+                         (when-not (empty? events)
```

Comment:
Is there actually a case where `retrieve-events` will return with an empty seq of events?